### ISSUE TITLE
Allow user to edit page or master contentlet

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#23225]
### Is your feature request related to a problem? Please describe.
As a dotCMS user, I want to be able to decide if I would like to edit a piece of content just for the page I'm editing or for the whole system.

### Describe the solution you'd like
When we know the contentlet is being used in more than one page, let's show the user a dialog, so it can choose if they want to edit the contentlet for the current page or for all the pages (master).

### Additional context
If the user selects "just for this page", we need to hit the endpoint to copy in place the contentlet and they open the editor for this one.

- [To know if the contentlet is being use in more than one page](https://github.com/dotcms/core/issues/23223)
- [To create a copy of the contentlet](https://github.com/dotcms/core/issues/23224)

### Mock
[Figma Prototype](https://www.figma.com/proto/trbwjLuf5F6KaU1rFucNeg/Portlets-%2F-Edit-Page?page-id=0%3A1&node-id=569%3A4774&viewport=4311%2C-717%2C0.42&scaling=min-zoom&starting-point-node-id=569%3A4774)

<img width="973" alt="image" src="https://user-images.githubusercontent.com/751424/220953680-d203a09d-aa95-494b-b625-f10578975b6f.png">

### Copy in place endpoint

```bash
curl --location --request PUT 'http://localhost:8080/api/v1/page/copyContent' \
--header 'Content-Type: application/json' \
--data-raw '{
    "containerId": "",
    "contentId": "",
    "pageId": "",
    "personalization": "",
    "relationType": "",
    "treeOrder": "",
    "variantId": ""
}'
```

### Acceptance criteria
- Show the dialog only when the content is used in multiples pages
- Allow the user to edit the contentlet for this page or all pages.

Updated by @fmontes

**Comment by fmontes** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Lunik-61780f8ea7866b001141b588/issues/dotCMS/core/23225?planning-poker) with Zenhub @hmoreras @jcastro-dotcms @jdotcms @rjvelazco

**Comment by fmontes** [comment]
Update the design here.

**Comment by jcastro-dotcms** [comment]
### INTERNAL QA: PASSED
Content Editors will now see a modal when trying to edit a contentlet that is being referenced in more than one page. The new design looks like this:

![edit-contentlet](https://user-images.githubusercontent.com/8779585/231203443-d9d82a8e-eb1a-4e63-a0da-9370b7dd5f1e.png)

**Comment by josemejias11** [comment]
Assigning to @bryanboza as per slack conversation.

**Comment by bryanboza** [comment]
After test some things we need some backend work here: 

- Fist of all we lost the property in the responses, now we are returning this just in the html as we can see in the follow image 
![Image](https://user-images.githubusercontent.com/2641437/233164601-b2944720-a71d-4d4e-9801-6eb4ccee9b45.png)

Then I assume we don't have a the correct postman test because this is not failing 

![Image](https://user-images.githubusercontent.com/2641437/233165967-71d8c186-7738-4d05-8c40-88897c84661a.png)

- The second part is that after you edit a piece of content that is used in two pages, and you select edit just for the selected page (copy and edit) we are not updating the reference and if you call the endpoint the previous content still coming with the property `on-number-of-pages= 2` then if you go to the other page that uses the same piece of content and try to edit, we still asking if want edit just for the page or for all the page. And the piece of content is just used in that page, wich does not make sense
Here a video to explain the case. 

https://user-images.githubusercontent.com/2641437/233169322-0d24cb8a-b460-47a1-b6ed-ef776086fff1.mov

**Comment by bryanboza** [comment]
Related to first point we seems we change this here https://github.com/dotCMS/core/issues/24217. 

Then maybe we need to doc this card. Tag Added

**Comment by rjvelazco** [comment]
### Passed Internal QA

Docker Image: [`dotcms/dotcms:23.05_849e8a4c_SNAPSHOT`]

https://user-images.githubusercontent.com/72418962/233494192-587de5ba-6cc0-47f8-8c7e-da5d0c6989ef.mov

**Comment by bryanboza** [comment]
Fixed, tested on release-23.05 // Docker // FF
